run: Add --uts flag

dcantah opened the pull request:
Adds support for specifying the host UTS namespace (the only thing
docker supported). This also consolidates the namespace options
checks for linux on run into a single function.

dcantah commented:
Unfortunately don't have a decent box to verify this for a day or two, so draft for now

dcantah commented:
Same strange (potential) rate-limit behavior on the project checks. If anyone can re-kick that off I'd appreciate

Zheaoli commented:
Thanks for your PR! In my personal opinion, I think it's unnecessary to change the `setPlatformOptions` signature in this PR. One PR for one thing, please.

If we need to change all signatures related to `internalLabels`, maybe we need another issue to discuss it.

dcantah commented:
It's not worth the hassle then don't worry about it, I'll move the signature change to another PR then. It was impacting the ability to just easily setup all the pid options enough it seemed warranted, and it's not exported or some relied on API here.

dcantah commented:
@Zheaoli, removed the setPlatformOptions signature change. I'll make an issue/separate PR

AkihiroSuda commented:
> @Zheaoli, removed the setPlatformOptions signature change. I'll make an issue/separate PR. Thanks!

Doesn't seem removed

AkihiroSuda commented:
A separate PR is preferred, but a separate commit in the same PR is also ok IMHO

dcantah commented:
> > @Zheaoli, removed the setPlatformOptions signature change. I'll make an issue/separate PR. Thanks!
> 
> Doesn't seem removed

It's the same signature still, it's just over multiple lines as it was a bit long. The signature change was:

### Before 
```go
func setPlatformOptions(
 	ctx context.Context,
 	opts []oci.SpecOpts,
 	cmd *cobra.Command,
 	client *containerd.Client,
 	id string,
 	internalLabels internalLabels,
 ) ([]oci.SpecOpts, internalLabels, error) {
```

### After 
```go
func setPlatformOptions(
 	ctx context.Context,
 	opts []oci.SpecOpts,
 	cmd *cobra.Command,
 	client *containerd.Client,
 	id string,
 	internalLabels *internalLabels,
 ) ([]oci.SpecOpts, error) {
```

but I reverted the `after` variant after the discussion

dcantah commented:
> A separate PR is preferred, but a separate commit in the same PR is also ok IMHO (for this amount of PR)

Yea, this didn't seem too unwieldy to all be in one PR, but I'll just make a separate at this point no worries.

dcantah commented:
Rebased after https://github.com/containerd/nerdctl/pull/1771 landed.

AkihiroSuda commented on cmd/nerdctl/run_linux.go:
```diff
@@ -136,17 +136,6 @@ func setPlatformOptions(
-	}
-	internalLabels.pidContainer = pidInternalLabel
-	opts = append(opts, pidOpts...)
-
```
The pidns changes seem irrelevant to utsns, could you consider splitting the commit? Not necessary to open a separate PR.

dcantah replied:
Sure, will try and get to today

dcantah replied:
@AkihiroSuda done

AkihiroSuda reviewed (approved):
Thanks
